commit d8f29266e99c7b82e4968cace1da1d16b3db06b5
Author: Stefan Kost <ensonic@users.sf.net>
Date:   Thu Feb 17 16:57:42 2011

    avidemux: also add the frame-type for the stream index

diff --git a/gst/avi/gstavidemux.c b/gst/avi/gstavidemux.c
index 0de3edf..321d216 100644
--- a/gst/avi/gstavidemux.c
+++ b/gst/avi/gstavidemux.c
@@ -4491,9 +4491,10 @@ gst_avi_demux_add_assoc (GstAviDemux * avi, GstAviStream * stream,
         GST_FORMAT_BYTES, offset, NULL);
     /* well, current_total determines TIME and entry DEFAULT (frame #) ... */
     gst_index_add_association (avi->element_index, stream->index_id,
-        GST_ASSOCIATION_FLAG_NONE,
-        GST_FORMAT_TIME, stream->current_total, GST_FORMAT_BYTES, offset,
-        GST_FORMAT_DEFAULT, stream->current_entry, NULL);
+        keyframe ? GST_ASSOCIATION_FLAG_KEY_UNIT :
+        GST_ASSOCIATION_FLAG_DELTA_UNIT, GST_FORMAT_TIME, stream->current_total,
+        GST_FORMAT_BYTES, offset, GST_FORMAT_DEFAULT, stream->current_entry,
+        NULL);
   }
 }
 
